lib/veteran_enrollment_system/associations/service.rb

```diff
@@ -189,6 +284,61 @@ def handle_ves_update_response(response, form_id)
           raise_error(response)
         end
       end
+
+      def get_address_from_association(association)
+        address = {}
+        fill_address_mappings_from_association(address, association)
+        fill_address_region_from_association(address, association)
+        address
+      end
+
+      def fill_address_mappings_from_association(address, association)
+        ADDRESS_MAPPINGS.each do |address_map|
+          address[address_map.first] = association['address'][address_map.last.to_s]
+        end
+      end
+
+      def fill_address_region_from_association(address, association)
+        case address['country']
+        when 'MEX'
+          fill_mexico_address_from_association(address, association)
+        when 'USA'
+          fill_usa_address_from_association(address, association)
+        else
+          fill_other_address_from_association(address, association)
+        end
+      end
+
+      def fill_mexico_address_from_association(address, association)
+        address['state'] = HCA::OverridesParser::STATE_OVERRIDES['MEX'].invert[address['state']]
+        address['postalCode'] = association['address']['postalCode']
+      end
+
+      def fill_usa_address_from_association(address, association)
+        address['state'] = association['address']['state']
+        zip_code = association['address']['zipCode']
+        zip_plus4 = association['address']['zipPlus4']
+        address['postalCode'] = zip_plus4.present? ? "#{zip_code}-#{zip_plus4}" : zip_code
+      end
+
+      def fill_other_address_from_association(address, association)
+        address['state'] = association['address']['provinceCode']
+        address['postalCode'] = association['address']['postalCode']
+      end
+
+      def fill_association_full_name_from_ves_association(association, ves_association)
+        NAME_MAPPINGS.each do |mapping|
+          association['fullName'][mapping.first] = ves_association['name'][mapping.last.to_s]
+        end
+      end
```

Comment:
We have very similar methods for the existing SOAP API that @dellerbie created. Once we transition over to the REST API, I'm assuming we'll only need to use these.